### PR TITLE
Feature/fnmoc

### PR DESCRIPTION
**Pull request by aerorahul**
add surface observations and ship track converters from GODAE.

### CODE REVIEW

File: src/marine/godae/trak2ioda.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        times.units = 'hours'
+        times.description = 'why does this have to be with a reference to, why not absolute?'
+
+        sal = ncid.createVariable('sea_surface_salinity@ObsValue', 'f4', ('nlocs',))
+        sal.units = 'g/kg'
+        sal.description = ''
+
+        sal_errs = ncid.createVariable(
+            'sea_surface_salinity@ObsError', 'f4', ('nlocs',))
+        sal_errs.units = '(g/kg)^2'
```
Comment (guillaumevernieres) [inline src/marine/godae/trak2ioda.py]: I think we need the standard deviation of the obs error (g/kg)

File: src/marine/godae/trak2ioda.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        u_errs.description = 'Standard deviation of observation error'
+
+        v = ncid.createVariable(
+            'sea_surface_meridional_wind@ObsValue', 'f4', ('nlocs',))
+        v.units = 'm/s'
+        v.description = ''
+
+        v_errs = ncid.createVariable(
+            'sea_surface_meridional_wind@ObsError', 'f4', ('nlocs',))
+        v_errs.units = '(m/s)^2'
```
Comment (guillaumevernieres) [inline src/marine/godae/trak2ioda.py]: same as above, std deviation instead of variance.

**Comment by aerorahul** [comment]
I will update the readme.

File: src/marine/godae/trak2ioda.py (hunk 2 above)
Comment (aerorahul) [reply src/marine/godae/trak2ioda.py]: I simply followed what is in profile2ioda.py.
I also noticed that in profile2ioda.py, someone added a floor value of 0.5 to the temperature obs. err.
The ob err. in track and surface obs. is currently arbitrary and needs to be addressed at some point.
At some point we will standardize these to use the netCDF API from @srherbener for writing to IODA format.

**Comment by travissluka** [comment]
should the appropriate CMakeLists.txt files be added so that the marine scripts get placed in the install directory like the other converters?

**Comment by aerorahul** [comment]
@travissluka That is a good idea.  I will update the PR, and ask for a re-review.

File: src/marine/godae/ship2ioda.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        idata is the IODA required data structure
+        '''
+
+        # idata is the dictionary containing IODA friendly data structure
+        idata = self.odata
+
+        # Diff. between ob. time and time in the file (hrs)!
+        for i in range(self.odata['n_obs']):
+            dtg = datetime.strptime(self.odata['ob_dtg'][i], '%Y%m%d%H%M')
+            idata['ob_dtg'][i] = (dtg - self.date).seconds/3600.
```
Comment (travissluka) [inline src/marine/godae/ship2ioda.py]: the wrong values are being produced, this should use **.total_seconds()** instead

File: src/marine/godae/trak2ioda.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        idata is the IODA required data structure
+        '''
+
+        # idata is the dictionary containing IODA friendly data structure
+        idata = self.odata
+
+        # Diff. between ob. time and time in the file (hrs)!
+        for i in range(self.odata['n_obs']):
+            dtg = datetime.strptime(self.odata['ob_dtg'][i], '%Y%m%d%H%M')
+            idata['ob_dtg'][i] = (dtg - self.date).seconds/3600.
```
Comment (travissluka) [inline src/marine/godae/trak2ioda.py]: use .total_seconds()

File: src/marine/godae/ship2ioda.py (hunk 3 above)
Comment (travissluka) [reply src/marine/godae/ship2ioda.py]: same thing for profile2ioda.py as well

Comment (travissluka) [reply src/marine/godae/ship2ioda.py]: ...noticeable if you pass a date into the script that is different from the base date of the obs file

File: src/marine/godae/profile2ioda.py [hunk 5]
```diff
@@ -158,7 +158,7 @@ def _to_nlocs(self):
             dtg = datetime.strptime(self.odata['ob_dtg'][n], '%Y%m%d%H%M')
 
             # Diff. between ob. time and time in the file (hrs)!
-            ddtg = (dtg - self.date).seconds/3600.
+            ddtg = (dtg - self.date).total_seconds/3600.
```
Comment (travissluka) [inline src/marine/godae/profile2ioda.py]: **total_seconds** is a function, needs **()**
same with other 2 locations

Comment (aerorahul) [reply src/marine/godae/profile2ioda.py]: thanks for the catch.

File: src/marine/godae/trak2ioda.py (hunk 4 above)
Comment (aerorahul) [reply src/marine/godae/trak2ioda.py]: done.

File: src/marine/godae/ship2ioda.py (hunk 3 above)
Comment (aerorahul) [reply src/marine/godae/ship2ioda.py]: indeed.

File: src/marine/godae/trak2ioda.py (hunk 2 above)
Comment (aerorahul) [reply src/marine/godae/trak2ioda.py]: fixed.

**Comment by aerorahul** [comment]
merged.

**Pull request merged**